### PR TITLE
Catch buildkite exception

### PR DESCRIPTION
**Pull request by patrobinson**
In the rare occasion the agent stops before we can stop it, spot_build will die unexpectedly

```
Traceback (most recent call last):
    22: from /usr/local/rbenv/versions/2.5.0/bin/spot_build:23:in `<main>'
    21: from /usr/local/rbenv/versions/2.5.0/bin/spot_build:23:in `load’
    20: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/bin/spot_build:5:in `<top (required)>'
    19: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build.rb:19:in `run’
    18: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build.rb:19:in `loop’
    17: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build.rb:20:in `block in run’
    16: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build.rb:20:in `each’
    15: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build.rb:21:in `block (2 levels) in run’
    14: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build/sqs_event.rb:19:in `shutdown_if_required’
    13: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build.rb:30:in `block (3 levels) in run’
    12: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build/buildkite_agent.rb:15:in `the_end_is_nigh’
    11: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spot_build-0.0.4/lib/spot_build/buildkite_agent.rb:21:in `stop’
    10: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/buildkit-0.4.0/lib/buildkit/client/agents.rb:60:in `stop_agent’
     9: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/buildkit-0.4.0/lib/buildkit/client.rb:59:in `put’
     8: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/buildkit-0.4.0/lib/buildkit/client.rb:109:in `request’
     7: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/sawyer-0.6.0/lib/sawyer/agent.rb:94:in `call’
     6: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.2/lib/faraday/connection.rb:177:in `put’
     5: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.2/lib/faraday/connection.rb:377:in `run_request’
     4: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.2/lib/faraday/rack_builder.rb:139:in `build_response’
     3: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.2/lib/faraday/response.rb:8:in `call’
     2: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.2/lib/faraday/response.rb:57:in `on_complete’
     1: from /usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.2/lib/faraday/response.rb:9:in `block in call’
/usr/local/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/buildkit-0.4.0/lib/buildkit/response/raise_error.rb:14:in `on_complete’: PUT https://api.buildkite.com/v1/organizations/envato-marketplaces/agents/[REDACTED]/stop: 422 - Can only stop connected agents (Buildkit::UnprocessableEntity)
Traceback (most recent call last):
```

This fixes that by swallowing the error and continuing on.

We also add some tests around this class while we're here.

In the future, this tool may be deprecated in favour of https://github.com/buildkite/lifecycled

### CODE REVIEW

File: lib/spot_build/buildkite_agent.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@ def the_end_is_nigh
     def stop(force="false")
       return unless agent_running?
       @client.stop_agent(@org_slug, agent_id, "{\"force\": #{force}}")
+    rescue Buildkit::UnprocessableEntity
+      # Swallow the error, this is generally thrown when the agent has already stopped
```
Comment (simpsora) [inline lib/spot_build/buildkite_agent.rb:23]: I'm not sure where (if anywhere) the output goes, but is there any value in emitting a warning message when this happens?

File: lib/spot_build/buildkite_agent.rb [hunk 2]
```diff
@@ -42,11 +44,11 @@ def job_build(build_url)
     end
 
     def current_job
-      agent[:job]
+      agent.job
```
Comment (simpsora) [inline lib/spot_build/buildkite_agent.rb:47]: Why has this syntax changed?

Comment (patrobinson) [reply lib/spot_build/buildkite_agent.rb:47]: To support the test doubles, which only allow the accessor method and not the hash access method.

Comment (simpsora) [reply lib/spot_build/buildkite_agent.rb:47]: Ah, right.  Carry on.

**Review by simpsora — APPROVED** [review]

File: lib/spot_build/buildkite_agent.rb (hunk 1 above)
Comment (patrobinson) [reply lib/spot_build/buildkite_agent.rb:23]: It goes to syslog, I don't think it's valuable for this particular exception but could be if some other unknown issue occurs?

**Pull request merged**